ui/pages/confirmations/components/confirm/info/approve/hooks/use-is-nft.ts

```diff
@@ -0,0 +1,23 @@
+import { TransactionMeta } from '@metamask/transaction-controller';
+import { useEffect, useState } from 'react';
+import { getTokenStandardAndDetails } from '../../../../../../../store/actions';
+
+export const useIsNFT = (
+  transactionMeta: TransactionMeta,
+): { isNFT: boolean } => {
+  const [decimals, setDecimals] = useState('');
+  useEffect(() => {
+    const fetchTokenDetails = async () => {
+      const result = await getTokenStandardAndDetails(
+        transactionMeta?.txParams?.to as string,
+      );
+
+      setDecimals(result?.decimals as string);
```

Comment:
Given the additional considerations of what to return while this request is in progress, or if the hook is unmounted, should we use `useAsyncResult` which handles all that for us?